server/src/profiles/dto/create-profile.dto.ts

```diff
@@ -0,0 +1,28 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { User } from '@prisma/client';
+
+export class CreateProfileDto {
+  @ApiProperty({ required: false, nullable: true })
+  age?: number;
+
+  @ApiProperty({ required: false, nullable: true })
+  bio?: string;
+
+  @ApiProperty({ required: false, nullable: true })
+  subjects?: string;
+
+  @ApiProperty({ required: false, nullable: true })
+  rate?: number;
+
+  @ApiProperty({ required: false, nullable: true })
+  phone?: string;
+
+  @ApiProperty({ required: false, nullable: true })
+  city?: string;
+
+  @ApiProperty({ required: false, nullable: true })
+  state?: string;
+
+  @ApiProperty()
+  user: User;
```

Comment:
Would it be "better" to use `userId` here instead of an entire user object?

Reply:
Obviously, yes. However, Prisma looks for a user object when creating a profile. which is why the create profile route is not a good way to generate profiles. profiles are instead created when a user is created. This route should never naturally be navigated to in the frontend anyway.